src/MongoDB/WriteConcern.c

```diff
@@ -406,6 +412,134 @@ static PHP_METHOD(WriteConcern, bsonSerialize)
 	convert_to_object(return_value);
 } /* }}} */
 
+/* {{{ proto string MongoDB\Driver\WriteConcern::serialize()
+*/
+static PHP_METHOD(WriteConcern, serialize)
+{
+	php_phongo_writeconcern_t* intern;
+	ZVAL_RETVAL_TYPE           retval;
+	php_serialize_data_t       var_hash;
+	smart_str                  buf = { 0 };
+	const char*                wtag;
+	int32_t                    w;
+	int32_t                    wtimeout;
+
+	intern = Z_WRITECONCERN_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (!intern->write_concern) {
+		return;
+	}
+
+	wtag     = mongoc_write_concern_get_wtag(intern->write_concern);
+	w        = mongoc_write_concern_get_w(intern->write_concern);
+	/* Note: PHP currently enforces that wimeoutMS is a 32-bit integer, so
+	 * casting will never truncate the value. This may change with PHPC-1411. */
+	wtimeout = (int32_t) mongoc_write_concern_get_wtimeout_int64(intern->write_concern);
```

Comment:
#1042 likely has implications for this, so whichever PR is merged first will require the other to be revised.

Since serialization is a case where we should definitely concern ourselves with portability between 32-bit and 64-bit architectures, this may warrant always serializing `wtimeout` values larger than `INT32_MAX` as an integer string and allowing `php_phongo_writeconcern_init_from_hash` to accept either type. Such an implementation would also allow us to write a single test that works on either architecture, since we could craft a serialized string that forces a `wtimeout` in the 64-bit range and then assert that it is printed accordingly by the debug handler (modified in #1042).

See [tests/functional/cursorid-001.phpt](https://github.com/mongodb/mongo-php-driver/blob/master/tests/functional/cursorid-001.phpt) for an example `EXPECTF` where we assert either string or integer output in a debug handler. I suppose an architecture-agnostic test for `wtimeout` could do the same.

Reply:
I'd like to get this merged and deal with 64-bit changes in #1042.